resource/config.go

```diff
@@ -25,8 +25,9 @@ type Config struct {
 	AssociatedResourceConfigs []AssociatedResourceConfig
 	Attributes                utils.AttributeMap
 
-	ConvertedAttributes ConfigValidator
-	ImplicitDependsOn   []string
+	AssociatedAttributes map[Name]AssociatedConfig
```

Comment:
I _sort_ of get the distinction between `AssociatedResourceConfigs` and `AssociatedResourceConfigs` but I'm gonna need a more clear explanation 😆 . I only ask because I'm not _fully_ convinced we need them both, and, if we do, those two names are confusing IMO. `AssociatedResourceConfig`s, to me, was the JSON boundary Go object for `service_configs`. Is `AssociatedAttributes` a "processed" version of that as a map of resource names to some other `AssociatedConfig`-fulfilling type that has an `Equals` method to be used in `Config.Equals` below? What's the problem with `reflect.DeepEqual`ing the `AssociatedResourceConfig`s?

Reply:
These are great questions.  I did this work to get something working but now that I'm going back with a more critical eye I'm not sure all this overhead is necessary.  Will investigate more

Reply:
I talked with @cheukt a bit about this and I dont think this will work.  `AssociatedResourceConfigs` are config parameters on the child resources the data service cares about and therefore diffing this way does not pick it up for purposes of data service config equality checks

Reply:
To clarify a bit more what these fields do AssociatedResourceConfigs is the raw json belonging to the resource that is a weak dependency of the data service.  The AssociatedAttributes field is the corollary to the `ConvertedAttributes` field in the config in that it is an object created from the raw json.  Cheuk and I think there has to be two separate fields for now. Very open to naming these whatever people think makes sense.  In general these various config fields confuse me as well so figuring out a clearer naming system would be beneficial.

Reply:
to add to it, I think one reason why I'm ok with doing Equals on AssociatedAttributes but not ConvertedAttributes is because the AssociatedAttributes is synthetic to begin with and is as close to a raw representation of a set of associated resource configs as we can get, whereas ConvertedAttributes should fall back to Attributes for the more original representation of the config